app/models/CategoryRecord.scala

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import cats.data.{EitherNec, NonEmptyChain}
+import cats.implicits.catsSyntaxTuple3Parallel
+import models.ott.CategorisationInfo
+import pages.{AssessmentPage, HasSupplementaryUnitPage, SupplementaryUnitPage}
+import play.api.libs.json.{Json, OFormat}
+import queries.RecordCategorisationsQuery
+
+final case class CategoryRecord(
+  eori: String,
+  recordId: String,
+  category: Int,
+  supplementaryUnit: Option[Int] = None,
+  measurementUnit: Option[String] = None
+)
+
+object CategoryRecord {
+
+  implicit lazy val format: OFormat[CategoryRecord] = Json.format
+
+  def build(answers: UserAnswers, eori: String, recordId: String): EitherNec[ValidationError, CategoryRecord] =
+    (
+      getCategory(answers, recordId),
+      answers.getOptionalPageValue(answers, HasSupplementaryUnitPage, SupplementaryUnitPage),
+      getMeasurementUnit(answers)
+    ).parMapN((category, supplementaryUnit, measurementUnit) =>
+      CategoryRecord(
+        eori = eori,
+        recordId = recordId,
+        category = category,
+        supplementaryUnit = supplementaryUnit,
+        measurementUnit = measurementUnit
+      )
+    )
+
+  def getCategory(answers: UserAnswers, recordId: String): EitherNec[ValidationError, Int] =
+    answers.get(RecordCategorisationsQuery) match {
```

Comment:
you can use a map here instead of matching on a some

Reply:
answers.get(RecordCategorisationsQuery).map(recordCategorisations => recordCategorisations.records.get(recordId)).map(categorisationInfo => chooseCategory(recordId, answers, categorisationInfo))

or better, put it in a for comprehension

```
for {
recordCategorisations  <- answers.get(RecordCategorisationsQuery)
...
categorisationInfo <- chooseCategory(recordId, answers, categorisationInfo)
yield result 
```